frontend/src/Pages/RoomExternalUrl.js

```diff
@@ -0,0 +1,180 @@
+                        navigator.clipboard.writeText(fullUrl);
+
+                        toast.success("Copied to clipboard!", {
+                            position: toast.POSITION.TOP_CENTER,
+                            autoClose: 3000,
+                            hideProgressBar: true
+                        });
+                    }} >
+                        <img className="w-full h-full" src={require("../assets/img/copy-regular.svg").default} />
+                        <ToastContainer />
+                    </button>
+                    <button className="w-5 h-5 mx-4" title="Replace" onClick={() => handleReplace(urlCode)} >
+                        <img className="w-full h-full" src={require("../assets/img/arrow-rotate-right-solid.svg").default} />
+                    </button>
+                </div>
+            </td>
+        </tr>
+    )
+}
+
+PermanentUrl.propTypes = {
+    urlCode: PropTypes.string,
+    handleReplace: PropTypes.func
+}
+
+const TemporaryUrl = ({ urlCode, handleDelete }) => {
+    const fullUrl = `https://matrix-cerberus/i/${urlCode}`;
+    return (
+        <tr className="h-20 text-sm leading-none text-gray-800 bg-white border-b border-t border-gray-100">
+            <td className="pl-4 cursor-pointer">
+                <p className='font-medium'> Temporary</p>
+            </td>
+            <td className="pl-12 cursor-pointer">
+                <p className="font-medium"> {urlCode} </p>
+            </td>
+            <td className="pl-12">
+                <p className="font-medium"> {fullUrl} </p>
+            </td>
+            <td className="pl-12">
+                <div className="flex justify-end">
+                    <button className="w-5 h-5 mx-4" title="Copy" onClick={() => {
+                        navigator.clipboard.writeText(fullUrl);
+
+                        toast.success("Copied to clipboard!", {
+                            position: toast.POSITION.TOP_CENTER,
+                            autoClose: 3000,
+                            hideProgressBar: true
+                        });
+                    }}>
+                        <img className="w-full h-full" src={require("../assets/img/copy-regular.svg").default} />
+                        <ToastContainer />
+                    </button>
+                    <button className="w-5 h-5 mx-4" title="Delete" onClick={() => handleDelete(urlCode)}>
+                        <img className="w-full h-full" src={require("../assets/img/delete-icon.svg").default} />
+                    </button>
+                </div>
+            </td>
+        </tr>
+    )
+}
+
+TemporaryUrl.propTypes = {
+    urlCode: PropTypes.string,
+    handleDelete: PropTypes.func
+}
+
+export default function RoomExternalUrl() {
+    const { roomId } = useParams();
+    const [permanentUrl, setPermanentUrl] = useState('');
+    const [temporaryUrl, setTemporaryUrl] = useState([]);
+
+    useEffect( () => {
+        async function fetchData() {
```

Comment:
Same feedback for all of these API calls, I would handle the error cases if you can.